src/io/leopard/toLeopard.ts

```diff
@@ -69,6 +69,215 @@ const JS_RESERVED_WORDS = [
+ *
+ * However, it *must* include even (capitalized) identifiers which *aren't*
+ * provided by Leopard, if any part of generated Leopard code could directly
+ * refer to those identifiers (expecting some browser-provided value, rather
+ * than a generated `Sprite` subclass!).
+ */
+const LEOPARD_RESERVED_SPRITE_NAMES = [
+  // Browser-provided identifiers
+  "Date",
+  "Math",
+
+  // Leopard-provided identifiers
+  "Color",
+  "Costume",
+  "Sound",
+  "Sprite",
+  "StageBase",
+  "Trigger",
+  "Watcher"
+];
+
+/**
+ * Property names which have special meaning in JavaScript. Custom properties
+ * must not overwrite these names, no matter the context.
+ */
+const JS_RESERVED_PROPERTIES = ["__proto__", "constructor", "prototype"];
+
+/**
+ * Property names which are used by any Leopard target - these correspond to
+ * Leopard's `SpriteBase` abstract class. Properties here are present on
+ * sprites as well as the stage.
+ *
+ * Overwriting these properties would change behavior that Leopard itself
+ * provides and expects to be behave in certain ways. While this a coding wizard
+ * might like to take advantage of this in their own Leopard project, generated
+ * projects must never accidentally overwrite these!
+ *
+ * This list is a superset of `JS_RESERVED_PROPERTIES` (and so are all supersets
+ * of this list).
+ *
+ * Note that this is *not* a superset of ordinary JavaScript reserved
+ * words. Properties are always accessed with `this.${name}` syntax, not used
+ * as standalone identifiers (`let ${name} = foo`).
+ */
+const LEOPARD_RESERVED_SPRITE_BASE_PROPERTIES = [
+  ...JS_RESERVED_PROPERTIES,
+
+  // Internals
+  "_costumeNumber",
+  "_layerOrder",
+  "_project",
+  "_vars",
+
+  // Basic execution
+  "triggers",
+  "vars",
+  "warp",
+
+  // Other objects
+  "stage",
+  "sprites",
+  "watchers",
+
+  // Control & events
+  "broadcast",
+  "broadcastAndWait",
+  "wait",
+
+  // Operators - casting
+  "toNumber",
+  "toBoolean",
+  "toString",
+  "compare",
+
+  // Operators - strings
+  "stringIncludes",
+  "letterOf",
+
+  // Operators - numbers
+  "degToRad",
+  "degToScratch",
+  "radToDeg",
+  "radToScratch",
+  "random",
+  "scratchTan",
+  "scratchToDeg",
+  "scratchToRad",
+  "normalizeDeg",
+  "wrapClamp",
+
+  // Lists (arrays)
+  "arrayIncludes",
+  "indexInArray",
+  "itemOf",
+
+  // Sensing
+  "answer",
+  "askAndWait",
+  "keyPressed",
+  "loudness",
+  "mouse",
+  "restartTimer",
+  "timer",
+
+  // Looks
+  "costume",
+  "costumeNumber",
+  "costumes",
+  "effects",
+
+  // Sounds
+  "audioEffects",
+  "effectChain",
+  "getSound",
+  "getSoundsPlayedByMe",
+  "playSoundUntilDone",
+  "sounds",
+  "startSound",
+  "stopAllOfMySounds",
+  "stopAllSounds",
+
+  // Pen
+  "clearPen"
+];
+
+/**
+ * Property names which are used by Leopard sprites (instances of `Sprite`,
+ * whether any subclass or directly constructed from `Sprite`). This list is
+ * a superset of `LEOPARD_RESERVED_SPRITE_BASE_PROPERTIES`.
+ */
```

Comment:
This comment is for sprites, but the code is for the stage. The comment below is also reversed.